lib/version_utils.pm

```diff
@@ -556,28 +557,52 @@ sub uses_qa_net_hardware {
     return !check_var("IPXE", "1") && check_var("BACKEND", "ipmi") || check_var("BACKEND", "generalhw");
 }
 
-=head2 get_sles_release
+=head2 get_os_release
 
-Get SLE release version and service spack info from any running sles os without any dependencies
+Get SLE release version, service pack and distribution name info from any running sles os without any dependencies
 It parses the info from /etc/os-release file, which can reside in any physical host or virtual machine
 The file can also be placed anywhere as long as it can be reached somehow by its absolute file path,
 which should be passed in as the second argument os_release_file, for example, "/etc/os-release"
 At the same time, connection method to the entity in which the file reside should be passed in as the
 firt argument go_to_target, for example, "ssh root at name or ip address" or "way to download the file"
 For use only on locahost, no argument needs to be specified
 =cut
-sub get_sles_release {
+sub get_os_release {
     my ($go_to_target, $os_release_file) = @_;
     $go_to_target    //= '';
     $os_release_file //= '/etc/os-release';
-    my $sles_release      = script_output("${go_to_target} cat ${os_release_file} | grep -i version= | grep -iEo \"[0-9]{1,}(\\\.|\\\-)?(sp)?([0-9]{1,})?\"");
-    my $sles_version      = '';
-    my $sles_service_pack = '';
-    my $auxiliary_var     = '';
-    ($sles_version,  $auxiliary_var)     = $sles_release =~ /^(\d+)[\.|\-]?(sp)?.*$/img;
-    ($auxiliary_var, $sles_service_pack) = $sles_release =~ /^${sles_version}[\.|\-]?(sp)?(\d+)$/img;
-    $sles_service_pack //= 0;
-    return $sles_version, $sles_service_pack;
+    my $os_distri       = script_output(sprintf('( . "%s" && echo $ID)', $os_release_file));
```

Comment:
hm I just realized that we "now" or maybe also before not using the "remote" `$os_release_file` but for `$os_distri` always the local one. Makes no sense for me, WDYT?

Reply:
I would like to make a proposal for this thing:

```
sub get_os_release {
    my ($go_to_target, $os_release_file) = @_;
    $go_to_target    //= '';
    $os_release_file //= '/etc/os-release';

    my %os_release = script_output ("$go_to_target cat $os_release_file") =~ /^(\S+)="?([^"\r\n]+)"?$/gm;
    %os_release = map { uc($_) => $pairs{$_}} keys %pairs;  # this is just for sanity cause you had grep -i version before

    my ($os_version, $os_service_pack) = split(/\.|-sp/i, $os_release{VERSION});
    $os_service_pack //= 0;
    return ($os_version, $os_service_pack, $os_release{ID});
}
```

Reply:
Thats looks really cool. I am stealing it. To be honest i was thinking to do something similar when i start the task but i got afraid to touch it. the reason and what i had in my mind was to make this map and let the callee ask for what he/she wants to return instead of having specific return values. Something like `get_os_release('VERSION_ID')` or get_os_release('PRETTY_NAME')`. In this way i would have avoid to write the new function i did. But then i would have to deal with all the modules that the virt team is using with this function.
WDYT? should i go an extra mile? maybe another ticket? or just apply this and we are good

Reply:
yeah maybe it would be good.
I would introduce a function like
```
sub read_key_value_file {
   my (%args) = @_;
   $args{go_to_target}    //= '';
    $args{file} //= '/etc/os-release';
    my %key_values = script_output ("$args{go_to_target} cat $args{file}_file") =~ /^(\S+)="?([^"\r\n]+)"?$/gm;
    return \%key_values;
}
```
And then just use it here and there. E.g. a `get_os_release_var('VERSION_ID');` could look like
```
sub get_os_release_var {
   my ($var_name, %args) = @_;
   return read_key_value_file(%args)->{$var_name};
}
```
WDYT?

Reply:
`read_key_value_file` is exactly what we need. but 

1. if we implement this it needs somethings extra to keep compatibility with the statements like [`($sles_running_version, $sles_running_sp) = get_os_release("ssh root\@$guest");`](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11223/files#diff-a092119d7e9e22c9d1a6ba66f49c86ff6fd4724f0a2e0909fcee929639e6907bR30) or 

2. keep  [first suggestion](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/11223#discussion_r509325575) but return `($os_version, $os_service_pack, $os_release{$var_name});` 

3. or we change as it was last suggested and we modify all the relevant modules which make use of it

I would rather avoid the number 3

Reply:
sure, lets go with option 2